From 361272e3063c11ce87dd655897f78a4c6aac81e8 Mon Sep 17 00:00:00 2001
From: 0x277F <0x277F@gmail.com>
Date: Tue, 28 Jun 2016 15:53:27 -0600
Subject: [PATCH] Changed server brand


diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index 87209d9..b6ebe74 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -783,7 +783,7 @@ public abstract class MinecraftServer
   }
 
   public String getServerModName() {
-    return "vanilla";
+    return "sink";
   }
 
   public CrashReport addServerInfoToCrashReport(CrashReport var1) {
-- 
2.7.4

