From afa077da08507ca19abce9c7fc2bac69ac010e55 Mon Sep 17 00:00:00 2001
From: Johannes Donath <johannesd@torchmind.com>
Date: Fri, 3 Jun 2016 05:36:14 +0200
Subject: [PATCH] Changed server brand.


diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index f1f2598..82a7ce5 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -781,7 +781,7 @@ public abstract class MinecraftServer
   }
 
   public String getServerModName() {
-    return "vanilla";
+    return "sink";
   }
 
   public CrashReport addServerInfoToCrashReport(CrashReport var1) {
-- 
2.7.1.windows.2

