From 14f5d8e262fcaa3c666c5ac6adc907770161b37a Mon Sep 17 00:00:00 2001
From: Johannes Donath <johannesd@torchmind.com>
Date: Fri, 16 Dec 2016 07:11:54 +0100
Subject: [PATCH] Changed Server Brand


diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index 87209d9..b6ebe74 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -783,7 +783,7 @@ public abstract class MinecraftServer
   }
 
   public String getServerModName() {
-    return "vanilla";
+    return "sink";
   }
 
   public CrashReport addServerInfoToCrashReport(CrashReport var1) {
-- 
2.11.0.windows.1

