From f4c1164edbe65f43d0055794ca27087317a0a1ca Mon Sep 17 00:00:00 2001
From: Johannes Donath <johannesd@torchmind.com>
Date: Fri, 3 Jun 2016 11:33:41 +0200
Subject: [PATCH] Eastereggs!


diff --git a/net/minecraft/entity/player/EntityPlayerMP.java b/net/minecraft/entity/player/EntityPlayerMP.java
index 848e056..63926ca 100644
--- a/net/minecraft/entity/player/EntityPlayerMP.java
+++ b/net/minecraft/entity/player/EntityPlayerMP.java
@@ -450,6 +450,38 @@ public class EntityPlayerMP extends EntityPlayer implements IContainerListener {
     this.addStat(StatList.DEATHS);
     this.takeStat(StatList.TIME_SINCE_DEATH);
     this.getCombatTracker().reset();
+
+    // BEE Start
+    if (!Boolean.getBoolean("org.basinmc.easteregg.disable")) {
+      // Akkarin's Easteregg - Imma spud
+      if (this.getUniqueID().toString().equals("d71a5dac-4e71-443b-8158-4389c269e44d")) {
+        ItemStack stack = new ItemStack(Items.POTATO, 1);
+        stack.setStackDisplayName("Akkarin's Spud");
+        stack.addEnchantment(net.minecraft.enchantment.Enchantment.getEnchantmentByID(51), 1);
+
+        this.dropItem(stack, true, false);
+      }
+
+      // __0x277F's Easteregg - Bacon!
+      if (this.getUniqueID().toString().equals("c10cd9e9-2c4a-49de-9f18-898cc9cfb9fe")) {
+        ItemStack stack = new ItemStack(Items.COOKED_PORKCHOP, 1);
+        stack.setStackDisplayName("__0x277F's Bacon");
+        stack.addEnchantment(net.minecraft.enchantment.Enchantment.getEnchantmentByID(51), 1);
+
+        this.dropItem(stack, true, false);
+      }
+
+      // Sycholic's Easteregg - Explodes
+      if (this.getUniqueID().toString().equals("f3a09056-082e-45ce-afd1-83e5214969af")) {
+        this.worldObj.createExplosion(this, this.posX, this.posY, this.posZ, 0f, false);
+      }
+
+      // Scruffy's Easteregg - Sponges!
+      if (this.getUniqueID().toString().equals("76a0fb2f-5db1-4346-8e35-7294cf625c38")) {
+        this.dropItem(new ItemStack(net.minecraft.init.Blocks.SPONGE, 1), true, false);
+      }
+    }
+    // BEE End
   }
 
   public boolean attackEntityFrom(DamageSource var1, float var2) {
-- 
2.11.0.windows.1

