From f7e3b90eb50e6198071270980881073f3534df46 Mon Sep 17 00:00:00 2001
From: 0x277F <0x277F@gmail.com>
Date: Tue, 28 Jun 2016 15:54:45 -0600
Subject: [PATCH] Enable access to the dedicated server settings


diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index b89e185..75b4cf1 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -472,6 +472,12 @@ public class DedicatedServer extends MinecraftServer implements IServer {
     this.saveProperties();
   }
 
+  // Basin Start
+  public PropertyManager getSettings() {
+    return this.settings;
+  }
+  // Basin End
+
   public boolean shouldBroadcastRconToOps() {
     return this.settings.getBooleanProperty("broadcast-rcon-to-ops", true);
   }
-- 
2.7.4

