From 028690683822213621a8e563a4777087d423370d Mon Sep 17 00:00:00 2001
From: 0x277F <0x277F@gmail.com>
Date: Sun, 19 Jun 2016 16:06:32 -0600
Subject: [PATCH] Expose MinecraftServer stopServer method


diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index 82a7ce5..046a013 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -340,7 +340,7 @@ public abstract class MinecraftServer
     }
   }
 
-  protected void stopServer() {
+  public void stopServer() { // Basin - private->public
     LOG.info("Stopping server");
     if (this.getNetworkSystem() != null) {
       this.getNetworkSystem().terminateEndpoints();
@@ -711,13 +711,13 @@ public abstract class MinecraftServer
       }
 
       var12.startServerThread();
-      Runtime.getRuntime()
+      /*Runtime.getRuntime()
           .addShutdownHook(
               new Thread("Server Shutdown Thread") {
                 public void run() {
                   var12.stopServer();
                 }
-              });
+              });*/ // Basin - Use our own shutdown hook.
     } catch (Exception var14) {
       LOG.fatal("Failed to start the minecraft server", var14);
     }
-- 
2.5.5

