From 404ce42e086a864d8c9fb0e5d8923b89746987a2 Mon Sep 17 00:00:00 2001
From: Johannes Donath <johannesd@torchmind.com>
Date: Wed, 21 Dec 2016 21:41:30 +0100
Subject: [PATCH] Removed the stock shutdown hook.


diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index 680d7a9..66a608f 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -719,13 +719,15 @@ public abstract class MinecraftServer
       }
 
       var12.startServerThread();
-      Runtime.getRuntime()
+      // Basin Start
+      /* Runtime.getRuntime()
           .addShutdownHook(
               new Thread("Server Shutdown Thread") {
                 public void run() {
                   var12.stopServer();
                 }
-              });
+              }); */
+      // Basin End
     } catch (Exception var14) {
       LOG.fatal("Failed to start the minecraft server", var14);
     }
-- 
2.13.0.windows.1

