From 9a81825b0433d2079f6f264200884361bb5a1e2b Mon Sep 17 00:00:00 2001
From: Johannes Donath <johannesd@torchmind.com>
Date: Thu, 22 Dec 2016 18:50:48 +0100
Subject: [PATCH] Disabled logging via System.out.


diff --git a/net/minecraft/init/Bootstrap.java b/net/minecraft/init/Bootstrap.java
index dede974..2bb98f4 100644
--- a/net/minecraft/init/Bootstrap.java
+++ b/net/minecraft/init/Bootstrap.java
@@ -487,7 +487,7 @@ public class Bootstrap {
   public static void register() {
     if (!alreadyRegistered) {
       alreadyRegistered = true;
-      redirectOutputToLog();
+      // Basin
       SoundEvent.registerSounds();
       Block.registerBlocks();
       BlockFire.init();
-- 
2.11.0.windows.1

